### DISCUSSION TITLE
what is the proper way of using a greeter?

### DISCUSSION THREAD

**Discussion by bruceloco** [WayfireWM/wayfire#2098] [Q&A]
Hi All

I am on my void-linux musl(because I like challenges or am stupid) on an ARM64 chromebook.
Not sure i understand how to use the greeter or how to start it.
I am using regreet.
I did the greetd configuration
the command is "wayfire -c path_to_config"
user greeter -> this always seems silly to me.
And in the wayfire config:
the plugin is autostart
in
[autostart] 
regreet=regreet -c /etc/greetd/regreet.toml
autostart_wf_shell=false

It will bring up wayfire with regreet - then I login, it closes the session and opens it with the correct user.
And doing this goes from tty at boot to wayland graphical mode to regreet to tty issuing some logging back to the user session I logged in with.
It is ugly to do it this way.
All graphical greeters require a wayland server to start up so they can work.
So I don't know how to do this

**Comment by soreau** [comment]
I think you have it setup mostly correctly since it's working. Whatever is changing the session doesn't seem to be doing it smoothly. Would it help if you started wayfire on a different tty than the greeter?

**Reply by BC79** [reply]
haven't tried that approach, is that a setting in wayfire? or would it be a command line option in the greeter to do a new tty session running wayfre?

**Comment by marcusbritanicus** [comment]
I have faced this issue at various stages. I believe this issue was raised with wlroots as well. In the meanwhile, this is what I have come up with:
1. Start greetd on a fresh tty. Typically, ttys 1-6 are occupied with console logins (agetty/agreety/etc..). So, use tty7, which is expected to be clean. You can do that by adding a section `[terminal]` to `/etc/greetd/config.toml` and adding `vt = 7` under that section. https://man.sr.ht/~kennylevinsen/greetd/#changing-target-vt
2. Redirect all the output from the greeter command to `/dev/null`. So instead of `command = "wayfire --config /path/to/wayfire.ini"` use: `command = "wayfire --config /path/to/wayfire.ini 2>&1 > /dev/null"` This will redirect all the wayfire output to `/dev/null` and keep the tty clean.

(I remember some discussion in wlroots regarding this transition from dm-wlroots-compositor to user-wlroots-compositor. Something related to sending blank frames to drm or similar, which would result in a smooth transition. But I have never seen a smooth transition with wayfire)

**Comment by BC79** [comment]
@marcusbritanicus Thank you for such a clear answer, this is very helpful.
I was following Kenny's instructions as I tried wlgreet.
In x86 systems this is all much smoother compared to Arm64 systems.
I think with Sway things work smoother, i haven't tested regreet with sway but that would be a 30 minute job.
Although it would be greeter->sway(greeter user)->regreet->sway(your user)
The other issue is that you can't exit a greeter. A logout sends you back to greet who sends you back to regreet.
There is no exit back to tty, which means for changes you have to do shutdowns.
